rest_framework/exceptions.py

```diff
@@ -216,8 +216,11 @@ def __init__(self, wait=None, detail=None, code=None):
             self.wait = None
         else:
             self.wait = math.ceil(wait)
-            self.detail += ' ' + force_text(ungettext(
-                self.extra_detail_singular.format(wait=self.wait),
-                self.extra_detail_plural.format(wait=self.wait),
-                self.wait
-            ))
+            code = self.detail.code
+            self.detail = ErrorDetail(
+                ' '.join((self.detail, force_text(ungettext(
+                    self.extra_detail_singular.format(wait=self.wait),
+                    self.extra_detail_plural.format(wait=self.wait),
+                    self.wait
+                )))))
+            self.detail.code = code
```

Comment:
Wondering if we could instead push this above the `super` call, and then just pass `detail` and `code` to that without having to explicitly construct an `ErrorDetail` instance. (Eg use a style more like`UnsupportedMediaType` above)